src/components/Wizard/CreateVmWizard/StorageTab.js

```diff
@@ -0,0 +1,404 @@
+      case ON_ACTIVATE: // new empty data
+        publishResults(rows, this.props.onChange, false); // data can't be valid ON_ACTIVATE
+        break;
+      case ON_CONFIRM: // new data
+      case ON_DELETE:
+      case ON_MOVE:
+        publishResults(rows, this.props.onChange, true); // expect data to be valid
+        break;
+      default:
+        break;
+    }
+  };
+
+  create = (renderConfig, values = {}) => {
+    this.setState(state => ({
+      nextId: state.nextId + 1,
+      rows: [
+        ...state.rows,
+        {
+          id: state.nextId,
+          isBootable: false,
+          editable: true,
+          edit: true, // trigger immediate edit
+          renderConfig,
+          ...values
+        }
+      ]
+    }));
+  };
+
+  createDisk = () => this.create(0);
+
+  attachStorage = () => this.create(1, { attachStorage: {} });
+
+  render() {
+    const columns = [
+      {
+        header: {
+          label: 'Disk Name',
+          props: {
+            style: {
+              width: '50%'
+            }
+          }
+        },
+        property: 'name',
+        hasAddendum: true,
+        renderConfigs: [
+          {
+            id: 'name-edit',
+            type: 'text'
+          },
+          {
+            id: 'name-attach-edit',
+            type: 'dropdown',
+            choices: this.props.storages.map(getName),
+            initialValue: '--- Select Storage ---'
+          }
+        ]
+      },
+      {
+        header: {
+          label: 'Size (GB)',
+          props: {
+            style: {
+              width: '23%'
+            }
+          }
+        },
+        property: 'size',
+        renderConfigs: [
+          {
+            id: 'size-edit',
+            type: 'positive-number'
+          }
+        ]
+      },
+      {
+        header: {
+          label: 'Storage Class',
+          props: {
+            style: {
+              width: '23%'
+            }
+          }
+        },
+        property: 'storageClass',
+
+        renderConfigs: [
+          {
+            id: 'storage-edit',
+            type: 'dropdown',
+            choices: this.props.storageClasses.map(getName),
+            initialValue: '--- Select ---'
+          }
+        ]
+      },
+      {
+        header: {
+          props: {
+            style: {
+              width: '4%'
+            }
+          }
+        },
+
+        renderConfigs: Array(3).fill({
+          id: 'actions',
+          type: ACTIONS_TYPE,
+          actions: [
+            {
+              actionType: DELETE_ACTION,
+              text: 'Remove Disk'
+            }
+          ],
+          visibleOnEdit: false
+        })
+      }
+    ];
+
+    // TODO: add create disk button once the create blank disk feature is implemented in kubevirt
+    /*
+     * <Button className="create-disk" onClick={this.createDisk} disabled={this.state.editing} id="create-disk-btn">
+     *   Create Disk
+     * </Button>
+    */
+    return (
+      <React.Fragment>
```

Comment:
is the TableFactory missing something ? Maybe you could use it and add missing stuff/improve it ?

Reply:
I think that much abstraction is not necessary here.